packages/react/modules/components/configure-adapter/configure-adapter.spec.js

```diff
@@ -15,21 +16,47 @@ const createTestProps = ({ adapter }) => ({
       key: 'foo-user-key',
     },
   },
+  adapterStatus: AdapterStates.CONFIGURED,
   onFlagsStateChange: jest.fn(),
   onStatusStateChange: jest.fn(),
   adapter,
 });
 
-const TestComponent = () => <span>Test component</span>;
+const TestComponent = props => {
+  const adapterContext = React.useContext(AdapterContext);
+  const isAdapterStatus = status => adapterContext.status === status;
+
+  return (
+    <>
+      <ul>
+        <li>
+          Is configuring:{' '}
+          {isAdapterStatus(AdapterStates.CONFIGURING) ? 'Yes' : 'No'}
+        </li>
+        <li>
+          Is configured:{' '}
+          {isAdapterStatus(AdapterStates.CONFIGURED) ? 'Yes' : 'No'}
+        </li>
+        <li>
+          Is unconfigured:{' '}
+          {isAdapterStatus(AdapterStates.UNCONFIGURED) ? 'Yes' : 'No'}
+        </li>
+      </ul>
+      {props.children}
+    </>
+  );
+};
 
 const render = ({ props, adapter }) => {
   const baseProps = createTestProps({ adapter });
   const mergedProps = { ...baseProps, ...props };
 
   const rendered = rtlRender(<ConfigureAdapter {...mergedProps} />);
-  const waitUntilReady = () => Promise.resolve();
 
-  return { ...rendered, waitUntilReady, props: mergedProps };
+  const waitUntilStatus = (status = AdapterStates.CONFIGURED) =>
+    rendered.findByText(`Is ${status}: Yes`);
```

Comment:
Maybe these should be waitUntilConfigured etc but it's easier this way. The downside is that rtl shows errors with non interpolate vairables.